sorc/tave.fd/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+list(APPEND fortran_src
+  tave.f
+)
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -i4 -r8")
+elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU)$")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fdefault-real-8")
+endif()
+
+set(exe_name tave.x)
+add_executable(${exe_name} ${fortran_src})
+target_link_libraries(
+  ${exe_name}
+  bacio::bacio_4
+  w3nco::w3nco_d
+  g2::g2_d)
```

Comment:
PNG, Jasper and ZLib are dependencies of g2.
When looking for g2, cmake ensures its (g2's) dependencies are found and included.
By large in cmake, one should only need to link to dependencies that the program/library uses.  It is the job of the dependencies to provide its own dependencies.

Reply:
That's what I expected. What about w3emc_libd? (I still assume it would've failed to build if needed.)

Reply:
I suspect it is not needed since I didn't see any calls to functions from that library.
If there was a test for this program (or any program for that matter), it would be easy to find out.